vyper/parser/parser_utils.py

```diff
@@ -449,26 +450,33 @@ def make_setter(left, right, location, pos, in_function_call=False):
             keyz = list(left.typ.members.keys())
         else:
             keyz = list(range(len(left.typ.members)))
+
+        # If the left side is a literal
+        if left.value == 'multi':
+            locations = [arg.location for arg in left.args]
+        else:
+            locations = [location for _ in keyz]
+
         # If the right side is a literal
         if right.value == "multi":
             if len(right.args) != len(keyz):
                 raise TypeMismatchException("Mismatched number of elements", pos)
             subs = []
-            for i, typ in enumerate(keyz):
-                subs.append(make_setter(add_variable_offset(left_token, typ, pos=pos), right.args[i], location, pos=pos))
+            for i, (typ, loc) in enumerate(zip(keyz, locations)):
+                subs.append(make_setter(add_variable_offset(left_token, typ, pos=pos), right.args[i], loc, pos=pos))
             return LLLnode.from_list(['with', '_L', left, ['seq'] + subs], typ=None)
         # If the right side is a null
         elif isinstance(right.typ, NullType):
             subs = []
-            for typ in keyz:
-                subs.append(make_setter(add_variable_offset(left_token, typ, pos=pos), LLLnode.from_list(None, typ=NullType()), location, pos=pos))
+            for typ, loc in zip(keyz, locations):
+                subs.append(make_setter(add_variable_offset(left_token, typ, pos=pos), LLLnode.from_list(None, typ=NullType()), loc, pos=pos))
             return LLLnode.from_list(['with', '_L', left, ['seq'] + subs], typ=None)
         # If tuple assign.
         elif isinstance(left.typ, TupleType) and isinstance(right.typ, TupleType):
             right_token = LLLnode.from_list('_R', typ=right.typ, location="memory")
             subs = []
             static_offset_counter = 0
-            for idx, (left_arg, right_arg) in enumerate(zip(left.args, right.typ.members)):
+            for idx, (left_arg, right_arg, loc) in enumerate(zip(left.args, right.typ.members, locations)):
                 # if left_arg.typ.typ != right_arg.typ:
```

Comment:
I believe this test can be removed? Do we have a test that checks types are correct?

Reply:
Yep! The type check is implicitly done in the recursive call to `make_setter`.

Reply:
Thought so :) just drop the comment and merge (sorry was a bit eager on the merge).